### ISSUE TITLE
Use `ESM` template by default instead of `CommonJS`

### ISSUE THREAD

**Issue by piranna** [fastify/fastify-cli#631]
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the feature has not already been requested

### 🚀 Feature Proposal

EcmaScript modules are supported by all maintained Node.js versions since long ago, so there's no need to use CommonJS modules anymore. I propose to make `--esm` flag enabled by default, and maybe also fully remove templates for CommonJS. Typescript templates would need to be updated too to generate ESM modules, but although tricky, it's a minor fix.

### Motivation

Embrace new and future prove standards, and deprecate old ones.

**Comment by mcollina** [comment]
While ESM module loading is stable, I don't think ESM is ready yet for general adoption due to the lack of stable loaders API. Take a look at https://github.com/nodejs/TSC/issues/1397.

**Comment by piranna** [comment]
> While ESM module loading is stable, I don't think ESM is ready yet for general adoption due to the lack of stable loaders API. Take a look at [nodejs/TSC#1397](https://github.com/nodejs/TSC/issues/1397).

You are right, I have had the same problems with loaders regarding Typescript + ESM combo, that's why I put the comment about it, but definitely it's a boomer.

**Comment by Tony133** [comment]
Closing this issue, as it dates back to 2023. For now, we want to give developers the freedom to choose whether to use CJS or ESM.

**Comment by piranna** [comment]
This is not about forcing users to use ESM, but instead change the default from CJS to ESM.
